operators/src/source/gdal_source.rs

```diff
@@ -456,7 +515,38 @@ where
         bbox: BoundingBox2D,
         spatial_resolution: SpatialResolution,
     ) -> BoxStream<Result<RasterTile2D<T>>> {
-        stream::iter(self.time_tile_iter(bbox, spatial_resolution))
+        // adjust the spatial resolution to the sign of the geotransform
+        let x_signed = if self
+            .dataset_information
+            .geo_transform()
+            .x_pixel_size
+            .is_sign_positive()
+            && spatial_resolution.x.is_sign_positive()
+        {
+            spatial_resolution.x
+        } else {
+            spatial_resolution.x * -1.0
+        };
+
+        let y_signed = if self
+            .dataset_information
+            .geo_transform()
+            .y_pixel_size
+            .is_sign_positive()
+            && spatial_resolution.y.is_sign_positive()
+        {
+            spatial_resolution.y
+        } else {
+            spatial_resolution.y * -1.0
+        };
+
+        let tiling_strategy = TilingStrategy {
+            bounding_box: bbox,
+            geo_transform: GeoTransform::new_with_coordinate_x_y(0.0, x_signed, 0.0, y_signed),
+            tile_pixel_size: [600, 600],
+        };
+
+        stream::iter(self.time_tile_iter(tiling_strategy))
```

Comment:
This stream is currently blocking, right?

Reply:
well this is an iterator which is always ready. so there is no blocking.

Reply:
But you don't spawn any thread for IO or am I missing something?

Reply:
load_tile_data_async uses tokio::task::spawn_blocking